### DISCUSSION TITLE
GitHub Action suddenly reports a syntax error on workflow file even though file wasn't changed

### DISCUSSION THREAD

**Discussion by laurent22** [community/community#55567] [Actions] [answered]
### Select Topic Area

Bug

### Body

I'm suddenly getting errors on all GitHub Actions, apparently due to a syntax error on a workflow even though that file hasn't changed in months. The error in on this line:

https://github.com/laurent22/joplin/actions/runs/4983319018/workflow#L22

<img width="508" alt="image" src="https://github.com/community/community/assets/1285584/7ec40551-22e3-4667-a929-d0c182b7a789">

I thought the lib I'm using (fkirc/skip-duplicate-actions@v5) may have changed but it didn't. I wouldn't even know how to fix a syntax error that's never been a syntax error. Any idea why it's doing this or what to do about it?

**Comment by laurent22** [comment]
Actually it seems some actions are still being executed successfully, so sometimes it's a syntax error, sometimes it's not... I'm sort of assuming it's yet another GitHub bug, I hope that's what it is anyway

**Comment by laurent22** [comment]
No, I spoke too quickly - all actions that run this particular workflow are failing. In fact, I can look back several weeks ago and all old actions are marked as failed too... **even though they succeeded back then**! What a complete non-sense. We are stuck at this point, any suggestions?

**Comment by airtower-luna** [comment, marked as answer]
I don't know why that worked, but [having multiple keys with they same name in a mapping is an error in YAML](https://yaml.org/spec/1.2.2/#mapping). My guess is that they updated the parser and it's a stricter now than it was before. I don't know what parser they use for Actions, but I've seen YAML parsers that'd simply use the last key and discard the previous ones. :sweat_smile: 

Or simply put: Having multiple `if`s for the same thing should have always been a syntax error. If you want to check two conditions, combine them with And (`&&`) like so:

```yaml
if: github.repository == 'laurent22/joplin' && needs.pre_job.outputs.should_skip != 'true'
```

**Comment by laurent22** [comment]
That makes perfect sense, thank you!
